DigiMeSDK/Core/Classes/Network/DMEAPIClient.m

```diff
@@ -159,8 +162,60 @@ - (void)requestFileWithId:(NSString *)fileId success:(void(^)(NSData *data))succ
     [self.queue addOperation:operation];
 }
 
-#pragma mark - Private
+#pragma mark - Data Push
 
+- (void)pushDataToPostboxWithPostboxId:(NSString *)postboxId
+                            sessionKey:(NSString *)sessionKey
+                   postboxRSAPublicKey:(NSString *)publicKey
+                        metadataToPush:(NSData *)metadata
+                            dataToPush:(NSData *)data
+                            completion:(PostboxDataPushCompletionBlock)completion
+{
+    DMEOperation *operation = [[DMEOperation alloc] initWithConfiguration:self.config];
+    
+    __weak __typeof(DMEOperation *) weakOperation = operation;
+    
+    operation.workBlock = ^{
+        
+        NSData *symmetricalKey = [self.crypto getRandomUnsignedCharacters:32];
+        NSData *iv = [self.crypto getRandomUnsignedCharacters:16];
+        NSString *metadataEncryptedString = [self.crypto preparePostboxMetadataWithKey:symmetricalKey initializationVector:iv metadata:metadata];
+        NSData *dataToPush = [self.crypto preparePostboxDataWithKey:symmetricalKey initializationVector:iv dataToPush:data];
+        NSString *keyEncrypted = [self.crypto preparePostboxSymmetricalKeyWithData:symmetricalKey rsaPublicKeyForEncryption:publicKey];
+        NSDictionary *headersToPush = [self postboxHeadersWithSessionKey:sessionKey symmetricalKey:keyEncrypted initializationVector:[iv hexString] metadata:metadataEncryptedString];
+        NSDictionary *headers = [self defaultPostboxHeaders];
```

Comment:
Is this method really ven needed? i noticed that in DMERequestFactory, you override the headers anyway.

You can create session like `NSURLSession *session = [self sessionWithHeaders:@{}];`